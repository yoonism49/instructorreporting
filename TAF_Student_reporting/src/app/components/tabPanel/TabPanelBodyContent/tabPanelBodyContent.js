// @flow weak

import React      from 'react';
import PropTypes  from 'prop-types';
import cx         from 'classnames';

const TabPanelBodyContent = ({
  id,
  isActive,
  children
}) => (
  <div
    id={id}
    className={
      cx({
        'tab-pane': true,
        active:  isActive
      })
    }>
    {children}
  </div>
);


TabPanelBodyContent.propTypes = {
  id:       PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  children: PropTypes.node.isRequired,
  isActive: PropTypes.bool
};

TabPanelBodyContent.defaultProps = {
  isActive: false
};

export default TabPanelBodyContent;
