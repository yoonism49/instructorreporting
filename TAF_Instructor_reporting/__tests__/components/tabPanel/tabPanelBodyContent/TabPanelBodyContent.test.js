
'use strict';

import React                from 'react';
import TabPanelBodyContent  from '../../../../src/app/components/tabPanel/tabPanelBodyContent/TabPanelBodyContent';
import renderer             from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme

describe('TabPanelBodyContent component', () => {
  it('renders as expected', () => {
    const mockProps = {
      id: 1,
      isActive: true
    };
    const component = renderer.create(
      <div>
        <TabPanelBodyContent { ...mockProps }>
          <p>TabPanelBody children here</p>
        </TabPanelBodyContent>
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
