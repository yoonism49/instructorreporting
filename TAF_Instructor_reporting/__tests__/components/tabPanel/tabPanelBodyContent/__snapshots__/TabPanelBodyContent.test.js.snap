// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanelBodyContent component renders as expected 1`] = `
<div>
  <div
    className="tab-pane active"
    id={1}
  >
    <p>
      TabPanelBody children here
    </p>
  </div>
</div>
`;
